### ISSUE TITLE
Dropbox node is broken after 0.112.0 update

### ISSUE THREAD

**Issue by arladmin** [n8n-io/n8n#1542]
**Describe the bug**
The Dropbox node gives an error:   "```Dropbox Credentials invalid!```" after updating n8n to 0.111.0

**To Reproduce**
1. Update n8n to the latest version (0.111.0)
2. Any existing workflows using the dropbox node will result in an Error 
3. Rolling back n8n to 0.110.3 resolves this issue

**Expected behavior**
There shouldn't have been any issue with Dropbox.

**Environment (please complete the following information):**
 - OS: [e.g. Ubuntu Linux 18.04] Ubuntu 20.04
 - n8n Version [e.g. 0.26.0] 0.111.0
 - Node.js Version [e.g. 10.16.0] 14.x

**Additional context**

**Comment by janober** [comment]
Can you please provide some more information like what resource and operations you are using and how you authenticate. Because I did just try the Dropbox node and it works perfectly fine for me.

**Comment by arladmin** [comment]
@janober  So far: 
Resource - File
Operation- Upload
Authentication- OAuth as well as Access Token

**Comment by RicardoE105** [comment]
Ok, we might have found the issue. In the meantime, can you recreate the credentials and please confirm that it solves the issue? Thanks.

**Comment by RicardoE105** [comment]
@arladmin, ok, we found the issue. We added a new scope needed to make an HTTP request that is used by all operations/resources. To make it work, connect the OAuth2 credentials again, and it should work as expected. We missed reflecting this in the breaking changes correctly, but I'm adding it right now. 

Thanks for letting us know.

**Comment by RicardoE105** [comment]
@arladmin The breaking change message was updated.

https://github.com/n8n-io/n8n/pull/1544/commits/c06971d510b1021baa8c4ba12155a608fabd43b8

**Comment by arladmin** [comment]
@RicardoE105  Still doesn't work.

Although, i had already tried initially to reconnect OAuth/reset OAuth credentials, i tried again with a different (local) instance.

Created a new workflow, added Dropbox node: File > Upload (using OAuth): same error.
Also tried with Access Token: same error.

![image](https://user-images.githubusercontent.com/64059154/111412192-fa5c4480-8701-11eb-974a-3a636ca93608.png)

Note: These credentials had been working fine in the earlier version.

**Comment by arladmin** [comment]
@RicardoE105  @janober  

Kindly reopen this issue as it's not yet resolved, as mentioned above.

**Comment by RicardoE105** [comment]
@arladmin, the only that comes to mind it's a mismatch between the scopes in the dropbox app and the scopes n8n is requesting. Can you please double-check that? Thanks

![image](https://user-images.githubusercontent.com/16496553/111561982-c453aa80-876b-11eb-9184-58b75a93bde0.png)

**Comment by arladmin** [comment]
@RicardoE105 

The credentials had already been operational beforehand, with the following scopes:

![image](https://user-images.githubusercontent.com/64059154/111564444-bc255a80-87bf-11eb-8807-ff681c3ba398.png)

The only thing that's working now is creating and using a new set of Dropbox OAuth Credentials in n8n with the exact same info as before. n8n just doesn't seem to accept the existing credentials, in this case.

**Comment by RicardoE105** [comment]
@arladmin, I was able to replicate the issue. I will investigate and keep you posted.

**Comment by arladmin** [comment]
@RicardoE105 

Another error propped up in the Dropbox node.

n8n version: 0.112.0

![image](https://user-images.githubusercontent.com/64059154/111899641-fa3aac80-8a53-11eb-9862-c3365f067cbf.png)

Kindly provide a resolution/workaround for this.

**Comment by jackdbd** [comment]
I have the same "Path root issue", both when authenticating using the OAuth credentials and the API access token. However the error occurs only on my VPS (a DigitalOcean droplet running a CapRover instance), so when n8n runs in a Docker container. If I run the very same workflow on my laptop with `n8n start` I get no errors and the workflow runs fine.

**Comment by arladmin** [comment]
@janober @RicardoE105 

Hi guys, is there any resolution/workaround available for this?

The Dropbox node is a pretty important node for us, that has now been dysfunctional for a week. 

Would appreciate if it can be repaired soon.

Thanks.

**Comment by RicardoE105** [comment]
@arladmin do not understand, I cannot replicate the issue. It's working fine for me. Can you please try that workflow below and tell me if it works for you. I wonder if it's something with the expression. 

```
{
  "nodes": [
    {
      "parameters": {},
      "name": "Start",
      "type": "n8n-nodes-base.start",
      "typeVersion": 1,
      "position": [
        250,
        300
      ]
    },
    {
      "parameters": {
        "path": "/asasas/dog.jpeg",
        "binaryData": true
      },
      "name": "Dropbox",
      "type": "n8n-nodes-base.dropbox",
      "typeVersion": 1,
      "position": [
        720,
        300
      ],
      "credentials": {
        "dropboxApi": "ASASASASA"
      }
    },
    {
      "parameters": {
        "url": "https://images.pexels.com/photos/1108099/pexels-photo-1108099.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500",
        "responseFormat": "file",
        "options": {}
      },
      "name": "HTTP Request",
      "type": "n8n-nodes-base.httpRequest",
      "typeVersion": 1,
      "position": [
        470,
        300
      ]
    }
  ],
  "connections": {
    "Start": {
      "main": [
        [
          {
            "node": "HTTP Request",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "HTTP Request": {
      "main": [
        [
          {
            "node": "Dropbox",
            "type": "main",
            "index": 0
          }
        ]
      ]
    }
  }
}
```

**Comment by arladmin** [comment]
@RicardoE105 

> @arladmin do not understand. I cannot replicate the issue. It's working fine for me. Can you please try that workflow below and tell me if it works for you. I wonder if it's something with the expression.
> 
> ```
> {
>   "nodes": [
>     {
>       "parameters": {},
>       "name": "Start",
>       "type": "n8n-nodes-base.start",
>       "typeVersion": 1,
>       "position": [
>         250,
>         300
>       ]
>     },
>     {
>       "parameters": {
>         "path": "/asasas/dog.jpeg",
>         "binaryData": true
>       },
>       "name": "Dropbox",
>       "type": "n8n-nodes-base.dropbox",
>       "typeVersion": 1,
>       "position": [
>         720,
>         300
>       ],
>       "credentials": {
>         "dropboxApi": "ASASASASA"
>       }
>     },
>     {
>       "parameters": {
>         "url": "https://images.pexels.com/photos/1108099/pexels-photo-1108099.jpeg?auto=compress&cs=tinysrgb&dpr=1&w=500",
>         "responseFormat": "file",
>         "options": {}
>       },
>       "name": "HTTP Request",
>       "type": "n8n-nodes-base.httpRequest",
>       "typeVersion": 1,
>       "position": [
>         470,
>         300
>       ]
>     }
>   ],
>   "connections": {
>     "Start": {
>       "main": [
>         [
>           {
>             "node": "HTTP Request",
>             "type": "main",
>             "index": 0
>           }
>         ]
>       ]
>     },
>     "HTTP Request": {
>       "main": [
>         [
>           {
>             "node": "Dropbox",
>             "type": "main",
>             "index": 0
>           }
>         ]
>       ]
>     }
>   }
> }
> ```

**No, this isn't working either.**

![image](https://user-images.githubusercontent.com/64059154/112085510-ac878680-8bb0-11eb-93a1-4459586ed09c.png)

**Comment by RicardoE105** [comment]
Ok, that makes it more strange cuz it works perfectly for me. It's really late for me right now. Do you have time to jump on a call tomorrow so that we can try to debug it?

**Comment by arladmin** [comment]
@RicardoE105 sure, later today, would appreciate it.

**Comment by RicardoE105** [comment]
Perfect, send me a private message to `Discourse.` My username is RicardoE105. I will send you my Calendly link so that you can pick a time,

**Comment by arladmin** [comment]
Adding this reference here, in case it might be relevant:

https://www.dropboxforum.com/t5/Dropbox-API-Support-Feedback/Download-on-scoped-app/m-p/462699/highlight/true#M23703

**Comment by jackdbd** [comment]
I created another Dropbox OAuth app with **full access** and it's working fine. The very same workflow does not work with a Dropbox OAuth app with **scoped access**. That `dropbox-api-path-root` seems indeed the issue.

**Comment by RicardoE105** [comment]
@jackdbd  @arladmin Yes, my app has Dropbox full access. 

I'm trying to find a way to not include the header `dropbox-api-path-root` when the app is type `App Folder`.

Added image for reference.

![image](https://user-images.githubusercontent.com/16496553/112155623-db572a00-8bbb-11eb-8006-8920f16085a1.png)

**Comment by RicardoE105** [comment]
@jackdbd @arladmin In the meantime, please use the Dropbox Full Access type while I figure something out.

**Comment by arladmin** [comment]
@RicardoE105 
Not really a feasible option, at the moment. Will wait further for the solution.

**Comment by RicardoE105** [comment]
@arladmin @jackdbd can you please test this PR locally and confirm it solves the issue. I added a new parameter in the credentials called `APP Access Type`. In your case, this parameter has to be set to App Folder.

**Comment by jackdbd** [comment]
I checked out this PR and now it's working fine for my Dropbox scoped app too. Thanks!

Also, while I was reading the Dropbox API docs I found that the tokens with no expiration will soon be deprecated. Maybe this could something useful to add to the node credentials documentation.
https://dropbox.tech/developers/migrating-app-permissions-and-access-tokens

> On September 30th, 2021, Dropbox will retire the creation of long-lived access tokens — resulting in all new tokens being short-lived.
